noir-projects/aztec-nr/aztec/src/utils/collapse.nr

```diff
@@ -0,0 +1,84 @@
+use dep::protocol_types::traits::Eq;
+
+// Collapses an array of Options with sparse Some values into a BoundedVec, essentially unwrapping the Options and
+// removing the None values. For example, given:
+//   input: [some(3), none(), some(1)]
+// this returns
+//   collapsed: [3, 1]
+pub fn collapse_array<T, let N: u32>(input: [Option<T>; N]) -> BoundedVec<T, N> where T: Eq {
```

Comment:
Renamed `collapse` function to `collapse_array` as `utils::collapse::colapse(...)` looked weird.